test/test_powersystemconstructors.jl

```diff
@@ -122,17 +122,43 @@ end
     # If that isn't appropriate for this type, add it to types_to_skip below.
 
     types_to_skip = (System,)
+    types_to_skip_set = (ThermalMultiStart,)
+    func_to_skip_set = (PowerSystems.set_dynamic_injector!,)
```

Comment:
Just curious, why is this not allowed for ThermalMultiStart but is allowed for other generators.  I'm assuming that the value for this field from the `demo` constructor will always be `nothing`.

Reply:
Not clear for me too. The error is basically that set_dynamic_injector in ThermalMultiStart is not implemented (but it is defined). The value is indeed nothing from the getter function. Here is the stack trace:
```
set_dynamic_injector! is not implemented for ThermalMultiStart
  Stacktrace:
   [1] error(::String) at ./error.jl:33
   [2] set_dynamic_injector!(::ThermalMultiStart, ::Nothing) at /Users/rhenriquez/.julia/dev/PowerSystems/src/models/injection.jl:12
```

Reply:
This is an issue with the fact that multistart doesn't have the proper field and should be added. But that should be done in a separate PR.

Reply:
The problem is that ThermalMultStart is not listed in the set_dynamic_injector! function in src/models/supplemental_accessors.jl.  You could add it to the Union in that function.  Maybe we should consider using ThermalGen and RenewableGen instead of concrete types.  I used concretes because I wasn't sure if every instance would have a dynamic generator.  I'll defer to you guys on this.

Your test found a bug!

Reply:
Agreed that should be based on the Abstract